src/pocketmine/network/mcpe/NetworkBinaryStream.php

```diff
@@ -74,6 +77,66 @@ public function putUUID(UUID $uuid) : void{
 		$this->putLInt($uuid->getPart(2));
 	}
 
+	public function getSkin() : Skin{
+		$skinId = $this->getString();
+		$skinResourcePatch = $this->getString();
+		$skinData = $this->getSkinImage();
+		$animationCount = $this->getLInt();
+		$animations = [];
+		for($i = 0; $i < $animationCount; ++$i){
+			$animations[] = new SkinAnimation(
+				$skinImage = $this->getSkinImage(),
+				$animationType = $this->getLInt(),
+				$animationFrames = $this->getLFloat()
+			);
+		}
+		$capeData = $this->getSkinImage();
+		$geometryData = $this->getString();
+		$animationData = $this->getString();
+		$premium = $this->getBool();
+		$persona = $this->getBool();
+		$capeOnClassic = $this->getBool();
+		$capeId = $this->getString();
+		$fullSkinId = $this->getString();
+
+		return new Skin($skinId, $skinData->getData(), $capeData->getData(), $skinResourcePatch, $geometryData);
```

Comment:
the geometry name should be extracted from the resource patch `default` field, as far as I understand.